docs/src/main/asciidoc/spanner.adoc

```diff
@@ -25,51 +27,180 @@ dependencies {
 }
 ----
 
+This setup takes care of bringing in the latest compatible version of Cloud Java Spanner libraries as well.
+
+== Spring Data Spanner
+
+
+=== Configuration
+
+To setup Spring Data Spanner, you have to configure connection to Cloud Spanner and Repositories (if you want to use them).
+
+==== Spanner settings
+
+You will need to use link:../spring-cloud-gcp-starters/spring-cloud-gcp-starter-data-spanner[Spring Boot Starter for Spring Data Spanner] to configure Google Cloud Spanner in your Spring application. It contains all the necessary setup that makes it easy to authenticate with your Google Cloud project.
+The following configuration options are available:
+
+|===
+| Name | Description | Optional | Default value
+| `spring.cloud.gcp.spanner.projectId` | the GCP project ID to use | Yes | provided by Spring Cloud GCP
+| `spring.cloud.gcp.spanner.instanceId` | Spanner instance to use | No |
+| `spring.cloud.gcp.spanner.database` |
+Spanner database to use | No |
+|===
+
+==== Repository settings
+
+Spring Data Repositories (see below for functionality) can be configured via the `@EnableSpannerRepositories` annotation on your main configuration file (typically the `@SpringBootApplication` file.
+
+==== Available beans
+
+With our Spring Boot autoconfiguration your application is able to inject an instance of `DatabaseClient` and `SpannerOperations`, and repositories are enabled, all the user defined repositories extending `SpannerRepository` are exposed as beans as well.
 
-=== Google Cloud Spanner
 
-Spring Data Spanner allows you to define entity classes and corresponding repository interfaces:
+=== Mapping POJO domain objects to Spanner tables using annotations
+
+Spring Data Spanner allows you to map POJOs to Spanner tables:
 
 [source,java]
 ----
-@SpannerTable(name = "students")
-public class SimpleStudent {
-	@SpannerPrimaryKeyColumn(keyOrder = 1)
-	String id;
+@Table(name = "trades")
+public class Trade {
+
+	@PrimaryKeyColumn(keyOrder = 2)
+	String action;
+
+	Double price;
 
-	@SpannerColumn(name = "first_name")
-	String name;
+	Double shares;
 
-	@SpannerColumnInnerType(innerType = String.class)
-	List<String> teachers;
+	@PrimaryKeyColumn(keyOrder = 1)
+	String symbol;
 
-	int age;
+	@Column(name = "trader_id")
+	String traderId;
+
+	@ColumnInnerType(innerType = Double.class)
+  List<Double> curve;
 }
 ----
 
-The `@SpannerTable` annotation provides the name of the Spanner table that stores instances of this class, one per row.
-The `@SpannerPrimaryKeyColumn` annotation is required and identifies the properties corresponding to the primary key columns in
+- `@Table` provides the name of the Spanner table that stores instances of this class, one per row.
```

Comment:
Don't know if this is what you intended, but these sentences are going to be rendered all in a row.
I think you might want to add a new line in between each sentence, otherwise it's going to be a long piece of text, hard to parse.

Reply:
not sure what you mean, these are rendering fine for me, did you have an issue with them? it's a list with `-` in the front.

Reply:
I think when I saw this for the first time, these sentences were all in a row. Nevermind.